### PR TITLE
feat: add docs generator

### PR DESCRIPTION
**Pull request by hampuslavin**
## Description
- Adds a docs generator that outputs one markdown file per command
- See example below for the cloud cli
image

## Tests
- See internal Slack for examples

### CODE REVIEW

File: lib/src/logger/loggers/void_logger.dart [hunk 1]
```diff
@@ -7,7 +7,7 @@ class VoidLogger extends Logger {
   VoidLogger() : super(LogLevel.debug);
 
   @override
-  int? get wrapTextColumn => null;
+  int? get wrapTextColumn => 80;
```
Comment (SandPod) [inline lib/src/logger/loggers/void_logger.dart:10]: Question: Why is this required for the void logger?

I don't think there is a guarantee that this will be set in a real environment so the code might need to handle this case.

File: test/documentation_generator/generate_markdown_test.dart [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  VegetableCommand() {
+    addSubcommand(AddVegetableCommand());
+  }
+
+  @override
+  void run() {}
+}
+
+void main() {
```
Comment (SandPod) [inline test/documentation_generator/generate_markdown_test.dart:83]: Question: Should we add a test that validates that multiple commands are supported and that each command ends up in its own file?

**Review by SandPod — APPROVED** [review]
LGTM! Given the comments are answered.

Could you update the description of this PR with the images from the slack channel as well?

Always helpful to get the full context when tracking changes through PRs in git.

File: test/documentation_generator/generate_markdown_test.dart (hunk 2 above)
Comment (hampuslavin) [reply test/documentation_generator/generate_markdown_test.dart:83]: I have this test already, is it enough? Or did you mean we should also verify the content in the other file?
```dart
 test('then outputs each command into a separate file', () {
      expect(output.keys, containsAll(['vegetable.md', 'spice.md']));
    });
   ```

File: lib/src/logger/loggers/void_logger.dart (hunk 1 above)
Comment (hampuslavin) [reply lib/src/logger/loggers/void_logger.dart:10]: Ah good catch, this was a left over from debugging 🙏

File: test/documentation_generator/generate_markdown_test.dart (hunk 2 above)
Comment (SandPod) [reply test/documentation_generator/generate_markdown_test.dart:83]: Aaah, that is probably enough! I missed that! 👍
